pkg/vault/contracts/VaultFactory.sol

```diff
@@ -2,28 +2,38 @@
 
 pragma solidity ^0.8.24;
 
-import { Authentication } from "@balancer-labs/v3-solidity-utils/contracts/helpers/Authentication.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
+
 import { IAuthorizer } from "@balancer-labs/v3-interfaces/contracts/vault/IAuthorizer.sol";
-import { CREATE3 } from "@balancer-labs/v3-solidity-utils/contracts/solmate/CREATE3.sol";
 import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
-import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {
+    ReentrancyGuardTransient
+} from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/ReentrancyGuardTransient.sol";
+import { CREATE3 } from "@balancer-labs/v3-solidity-utils/contracts/solmate/CREATE3.sol";
 
 import { ProtocolFeeController } from "./ProtocolFeeController.sol";
 import { VaultExtension } from "./VaultExtension.sol";
 import { VaultAdmin } from "./VaultAdmin.sol";
 
 /// @notice One-off factory to deploy the Vault at a specific address.
-contract VaultFactory is Authentication {
+contract VaultFactory is ReentrancyGuardTransient, Ownable2Step {
     bytes32 public immutable vaultCreationCodeHash;
     bytes32 public immutable vaultAdminCreationCodeHash;
     bytes32 public immutable vaultExtensionCreationCodeHash;
 
+    mapping(address vaultAddress => ProtocolFeeController) public protocolFeeController;
+    mapping(address vaultAddress => VaultExtension) public vaultExtension;
+    mapping(address vaultAddress => VaultAdmin) public vaultAdmin;
+    mapping(address vaultAddress => bool deployed) public isDeployed;
```

Comment:
```suggestion
    mapping(address vaultAddress => ProtocolFeeController) public deployedProtocolFeeControllers;
    mapping(address vaultAddress => VaultExtension) public deployedVaultExtensions;
    mapping(address vaultAddress => VaultAdmin) public deployedVaultAdmins;
    mapping(address vaultAddress => bool deployed) public deployedVaults;
```
Consider making the names clearer. (And then you wouldn't have all the name clashes with `protocolFeeController_`, `vaultExtension_`, etc.)